waddrmgr/address.go

```diff
@@ -263,10 +287,26 @@ func (a *managedAddress) ExportPrivKey() (*btcutil.WIF, error) {
 	return btcutil.NewWIF(pk, a.manager.chainParams, a.compressed)
 }
 
+// newManagedPublicAddress returns a new address based on the passed account
+// and public key hash.
+func newManagedPublicAddress(m *Manager, account uint32, pkHash []byte) (*managedAddress, error) {
```

Comment:
Hmm, how about `newManagedPKHashAddress`? WatchingOnly could mean addresses imported using `importpubkey` i.e. pubkey based addresses. This func is specifically for P2PKH addresses imported using just the pkhash.